paddle/fluid/operators/batch_norm_op.cc

```diff
@@ -428,12 +440,15 @@ class BatchNormGradKernel<platform::CPUDeviceContext, T>
     const float epsilon = ctx.Attr<float>("epsilon");
     const DataLayout data_layout =
         framework::StringToDataLayout(data_layout_str);
+    bool is_inplace = (x == y);
```

Comment:
Do not use `x == y` to distinguish whether in-place happens. Try to use `x_data == y_data`, because in-place pass would not change variable names and would share tensor buffers now.

Reply:
ok